hub/api/dataset.py

```diff
@@ -85,14 +85,7 @@ def __init__(
         self.index = index
 
         self.tensors: Dict[str, Tensor] = {}
-        if dataset_exists(self.storage):
-            self.meta = DatasetMeta.load(self.storage)
-            for tensor_name in self.meta.tensors:
-                self.tensors[tensor_name] = Tensor(tensor_name, self.storage)
-        elif len(self.storage) > 0:
-            raise PathNotEmptyException
-        else:
-            self.meta = DatasetMeta.create(self.storage)
+        self._load_meta()
```

Comment:
instead of `self._load_meta()` you should add a new property to the dataset. for reference, check out [how `Tensor` does this](https://github.com/activeloopai/Hub/blob/a6333eebead081fe0fff2a4c0c70e7cef0343474/hub/api/tensor.py#L87)

here's the method body:

```python
class Dataset:
    ...

    @property
    def meta(self):
        return DatasetMeta.load(self.storage)

    ...
```

Reply:
this won't slow things down because dataset will use cache

Reply:
and now every time you do `ds.meta` you will get the meta always

Reply:
this was needed to force reload the meta after transform and cache clearing. it's used in transform.py as well.

Reply:
ok, well can you at least use the `@property` and rename from `_load_meta` -> `meta`? that way it is always doing this loading? i don't want to have to remember to call this method and presumably the user will be affected by this also

Reply:
and given the normal case is non-cache reloading this shouldn't slow anything down